src/asm/parser.y

```diff
@@ -579,13 +579,8 @@ assert		: T_POP_ASSERT assert_type relocexpr
 		}
 ;
 
-shift		: T_POP_SHIFT		{ macro_ShiftCurrentArgs(); }
-		| T_POP_SHIFT uconst
-		{
-			int32_t i = $2;
-			while (i--)
-				macro_ShiftCurrentArgs();
-		}
+shift		: T_POP_SHIFT		{ macro_ShiftCurrentArgs(1); }
+		| T_POP_SHIFT uconst	{ macro_ShiftCurrentArgs($2); }
```

Comment:
If we're handling the loops directly, it might be better to accept any constant, and perform the sign check semantically, so that we the error message can be more descriptive.

Also, maybe a test case?

Reply:
Done.